net/sslh/src/opnsense/mvc/app/controllers/OPNsense/Sslh/forms/settings.xml

```diff
@@ -0,0 +1,197 @@
+<form>
+    <field>
+        <id>settings.enabled</id>
+        <label>Enable SSLH</label>
+        <type>checkbox</type>
+        <help><![CDATA[
+            This will enable the SSLH service.
+            ]]>
+        </help>
+    </field>
+    <field>
+        <id>settings.listen_addresses</id>
+        <label>Listen Addresses</label>
+        <help><![CDATA[
+            <b>-p, --listen</b>  address:port<br>
```

Comment:
further problems for translations: we don't add strings that offer markup since it could break the html/js, is prone to XSS. Markup and layout inside translations remain a huge problem with no easy solution: https://github.com/opnsense/lang/issues/1

Reply:
This makes me cry a little bit. I did this because it looks nice in the XML, and presents well on the page. I understand though. My first question, to which the answer is: You can't, is how do I do paragraphs?

I'd like to better understand this challenge, as I'd like to help if I can. I've seen the usage of gettext and _lang, but I don't see either being used here. I don't understand exactly how the translations work, any help there would be welcome.

I chose this style for the help messages because I've come across many help messages for settings in OPNsense where I could only assume that a particular setting is related to a setting in a service or application. Having this information available for anyone who is familiar with the application would be really helpful especially when it comes to looking up those settings in application documentation. OpenVPN was one standout experience I had with this because many of the settings are short-handed, but long in the UI, so it makes cross referencing difficult.

Not for here to discuss, but an idea which may help with the translation issues, utilizing pre defined styles
```xml
<help>
  <header>-p, --listen address:port</header>
  <body>Paragraph one.</body>
  <body>Paragraph two.</body>
</help>
```
Then do styling using the layout_partials. Would have to figure out which styles are useful, but that would be the basic idea. Can't do it with the current `parseFormNode()` but I've written one that can.

Reply:
> You can't, is how do I do paragraphs?

You can, but we won't translate it for the mentioned reasons.

> I'd like to better understand this challenge, as I'd like to help if I can.

Model strings are dynamically generated from XML content via https://github.com/opnsense/lang/blob/master/Scripts/plugins/models.py and automatically redirected through gettext() upon form generation. This is more of an oddity of how gettext works and does not work.

> I chose this style for the help messages because I've come across many help messages for settings in OPNsense where I could only assume that a particular setting is related to a setting in a service or application. 

Essentially you are writing inline documentation in a restricted scope (time and space). Most of elaborate documentation should be on docs.git and it's actually better to have one document for each plugin as we try with the ones that we as a core team introduce nowadays. It's not a bad thing, but having a shorter help in the GUI and a full documentation indexable by search engines has multiple advantages already: less strain on translations, simpler GUI, more time for users to do what they want to do while working with the GUI, full text search from search engine and documentation, examples and tutorials in documentation, more visibility for non-OPNsense users on the topic of the plugin.

> Not for here to discuss, but an idea which may help with the translation issues, utilizing pre defined styles

That looks like what we had in mind. The downside (I'm sorry to play devil's advocate) is that while the string are now clean and well-formed we essentially end up with splitting a block of translation that belongs together into separate strings that may or may not be translated, may or may not be translated by the same persons at the same time. Context could get lost. That's why for the most part we would prefer the documentation over this.

It would be nice if you could dump your idea in that lang.git issue for further discussion. Thanks :)

Reply:
I recently dug into the opnsense/lang repo while researching something else, and I believe I've gained a better understanding of the challenge. I've changed them to just a plain text description.

You mentioned the docs git, and I hadn't considered writing plugin docs there. I'm thinking the README I've put together will be a good base for such documentation. I'm guessing this is where it would live? https://github.com/opnsense/docs/tree/master/source/manual/how-tos